include/OpenSpaceToolkit/Astrodynamics/Access/Constraint.hpp

```diff
@@ -0,0 +1,88 @@
+/// Apache License 2.0
+
+#ifndef __OpenSpaceToolkit_Astrodynamics_Access_Constraint__
+#define __OpenSpaceToolkit_Astrodynamics_Access_Constraint__
+
+#include <optional>
+#include <variant>
+
+#include <OpenSpaceToolkit/Core/Container/Map.hpp>
+#include <OpenSpaceToolkit/Core/Type/Real.hpp>
+
+#include <OpenSpaceToolkit/Mathematics/Object/Interval.hpp>
+#include <OpenSpaceToolkit/Mathematics/Object/Vector.hpp>
+
+#include <OpenSpaceToolkit/Physics/Coordinate/Spherical/AER.hpp>
+
+namespace ostk
+{
+namespace astrodynamics
+{
+namespace access
+{
+
+using ostk::core::container::Map;
+using ostk::core::type::Real;
+
+using ostk::mathematics::object::Interval;
+using ostk::mathematics::object::Vector2d;
+
+using ostk::physics::coordinate::spherical::AER;
+
+class Constraint
```

Comment:
Comment on the naming, not sure I'm a huge fan of "Constraint for this class' role.
What about "Visibility" or "Window" or something like that?